<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="apriltag_settings" default="$(find tj2_bringup)/config/diffyjr/tags/settings.yaml"/>
    <arg name="apriltag_tags" default="$(find tj2_bringup)/config/diffyjr/tags/tags.yaml"/>
    <arg name="publish_tag_detections_image" default="false"/>
    <arg name="filter_params" default="$(find tj2_bringup)/config/diffyjr/northstar/filter/filter.yaml"/>
    <arg name="camera_0_params" default="$(find tj2_bringup)/config/diffyjr/northstar/camera/camera_0.yaml"/>
    <arg name="camera_1_params" default="$(find tj2_bringup)/config/diffyjr/northstar/camera/camera_1.yaml"/>

    <group ns="northstar">
        <rosparam command="load" file="$(arg filter_params)" ns=""/>

        <node pkg="tj2_northstar" type="tj2_northstar_filter.py" name="tj2_northstar" output="screen">
            <remap from="odom" to="/tj2/odom" />
        </node>

        <node pkg="tj2_northstar" type="landmark_converter.py" name="landmark_converter" required="false" output="screen"/>

        <!-- <node name="camera_0" pkg="usb_cam" type="usb_cam_node" output="screen" >
            <rosparam command="load" file="$(arg camera_0_params)" ns=""/>
            <remap from="image" to="camera_0/image_raw"/>
        </node>

        <node name="camera_1" pkg="usb_cam" type="usb_cam_node" output="screen" >
            <rosparam command="load" file="$(arg camera_1_params)" ns=""/>
            <remap from="image" to="camera_1/image_raw"/>
        </node> -->

        <include file="$(find tj2_northstar)/launch/apriltag_single_cam.launch">
            <arg name="apriltag_settings" value="$(arg apriltag_settings)"/>
            <arg name="apriltag_tags" value="$(arg apriltag_tags)"/>
            <arg name="node_name" value="apriltag_camera_0"/>
            <arg name="camera_info_topic" value="camera_0/camera_info"/>
            <arg name="image_rect_topic" value="camera_0/image_raw"/>
        </include>
        <include file="$(find tj2_northstar)/launch/apriltag_single_cam.launch">
            <arg name="apriltag_settings" value="$(arg apriltag_settings)"/>
            <arg name="apriltag_tags" value="$(arg apriltag_tags)"/>
            <arg name="node_name" value="apriltag_camera_1"/>
            <arg name="camera_info_topic" value="camera_1/camera_info"/>
            <arg name="image_rect_topic" value="camera_1/image_raw"/>
        </include>
    </group>
</launch>
